components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -71,21 +89,21 @@
             while (rs.next()) {
                 IdentityProvider identityProvider = new IdentityProvider();
                 identityProvider.setIdentityProviderName(rs.getString(1));
-                if (rs.getString(2).equals("1")) {
+                if (("1").equals(rs.getString("IS_PRIMARY"))) {
```

Comment:
Might be a good idea to introduce appropriate constants upon magic numbers like "1" in order to improve the maintainability of the code.